packages/generator-single-spa/src/vue/generator-single-spa-vue.js

```diff
@@ -20,49 +20,34 @@ module.exports = class SingleSpaVueGenerator extends Generator {
     });
   }
   async getOptions() {
-    while (!this.options.orgName) {
-      let { orgName } = await this.prompt([
-        {
-          type: "input",
-          name: "orgName",
-          message: "Organization name (use lowercase and dashes)",
-        },
-      ]);
+    const { dir, name } = path.parse(path.resolve(this.options.dir));
+    const hasValidName = validate(name) === true;
 
-      orgName = orgName && orgName.trim();
-      if (!orgName) {
-        console.log(chalk.red("orgName must be provided!"));
-      } else if (!isValidName(orgName)) {
-        console.log(chalk.red("orgName must use lowercase and dashes!"));
-      } else {
-        this.options.orgName = orgName;
-      }
-    }
-
-    let { dir, name } = path.parse(path.resolve(this.options.dir));
+    const { orgName, projectName = name } = await this.prompt([
+      {
+        type: "input",
+        name: "orgName",
+        message: "Organization name",
+        suffix: " (can use lowercase letters, numbers, dash or underscore)",
+        when: !this.options.orgName,
+        validate,
+      },
+      {
+        type: "input",
+        name: "projectName",
+        message: "Project name",
+        suffix: " (can use lowercase letters, numbers, dash or underscore)",
+        when: !hasValidName,
+        validate,
+      },
+    ]);
 
-    if (!isValidName(name)) {
-      while (!this.options.projectName) {
-        let { projectName } = await this.prompt([
-          {
-            type: "input",
-            name: "projectName",
-            message: "Project name (use lowercase and dashes)",
-          },
-        ]);
+    this.options.orgName = orgName;
+    this.options.projectName = projectName;
+    if (hasValidName) this.options.dir = dir;
 
-        projectName = projectName && projectName.trim();
-        if (!projectName) {
-          console.log(chalk.red("projectName must be provided!"));
-        } else if (!isValidName(projectName)) {
-          console.log(chalk.red("projectName must use lowercase and dashes!"));
-        } else {
-          this.options.projectName = projectName;
-        }
-      }
-    } else {
-      this.options.dir = dir;
-      this.options.projectName = name;
+    if (!fs.existsSync(this.options.dir)) {
+      fs.mkdirSync(this.options.dir);
```

Comment:
This fixed a bug I encountered where the vue-cli was failing silently if the directory did not exist.